### DISCUSSION TITLE
Container crashes while downloading GeoLite2

### DISCUSSION THREAD

**Discussion by dd-gamer** [shlinkio/discussions#85] [Help wanted] [answered]
I'm running Shlink in docker with these settings:
docker run -d \
	--name shlink \
	-p 8070:8080 \
	-e DEFAULT_DOMAIN=<my domain> \
	-e IS_HTTPS_ENABLED=true \
	-e GEOLITE_LICENSE_KEY=<my key>\
	--security-opt seccomp=unconfined \
	shlinkio/shlink:latest

When I leave the key entry empty, everything works fine, with the key, the container spins up and crashes immediately.
I tried both key versions provided by maxmind.

The console log:

Creating fresh database if needed...
Updating database...
Generating proxies...
Clearing entities cache...

If you can tell me how to enable debug mode to get additional information, please do so.

**Comment by acelaya** [comment]
Try providing `-e SHELL_VERBOSITY=3`. It should generate a more verbose output.

**Comment by dd-gamer** [comment]
In GeolocationDbUpdateFailedException.php line 34:
                                                                               
  [Shlinkio\Shlink\CLI\Exception\GeolocationDbUpdateFailedException]           
  An error occurred while updating geolocation database, and an older version  
   could not be found.                                                         
                                                                               
Exception trace:
  at /etc/shlink/module/CLI/src/Exception/GeolocationDbUpdateFailedException.php:34
 Shlinkio\Shlink\CLI\Exception\GeolocationDbUpdateFailedException::withoutOlderDb() at /etc/shlink/module/CLI/src/Util/GeolocationDbUpdater.php:106
 Shlinkio\Shlink\CLI\Util\GeolocationDbUpdater->downloadNewDb() at /etc/shlink/module/CLI/src/Util/GeolocationDbUpdater.php:55
 Shlinkio\Shlink\CLI\Util\GeolocationDbUpdater->downloadIfNeeded() at /etc/shlink/module/CLI/src/Util/GeolocationDbUpdater.php:43
 Shlinkio\Shlink\CLI\Util\GeolocationDbUpdater->checkDbUpdate() at /etc/shlink/module/CLI/src/Command/Visit/DownloadGeoLiteDbCommand.php:50
 Shlinkio\Shlink\CLI\Command\Visit\DownloadGeoLiteDbCommand->execute() at /etc/shlink/vendor/symfony/console/Command/Command.php:308
 Symfony\Component\Console\Command\Command->run() at /etc/shlink/vendor/symfony/console/Application.php:1002
 Symfony\Component\Console\Application->doRunCommand() at /etc/shlink/vendor/symfony/console/Application.php:299
 Symfony\Component\Console\Application->doRun() at /etc/shlink/vendor/symfony/console/Application.php:171
 Symfony\Component\Console\Application->run() at /etc/shlink/bin/cli:10
In DbUpdateException.php line 36:
                                                                               
  [Shlinkio\Shlink\IpGeolocation\Exception\DbUpdateException]                  
  An error occurred while trying to download a fresh copy of the GeoLite2 dat  
  abase



(Oh, and how can I make the url-shorts persistent, without using an extra database? What do I need to map?)

**Comment by acelaya** [comment]
This error means you did not provide a valid GeoLite2 license key.

> can I make the url-shorts persistent, without using an extra database

You can't. A database is mandatory.

**Comment by dd-gamer** [comment, marked as answer]
Found the issue, he was not able to resolve the dns name. I now mapped a custom resolv.conf, so that he uses another dns server. Problem solved.

**Comment by dd-gamer** [comment]
One more question, where is the API Token stored? It seems currently that I have to generate a new one for the webinterface, everytime I restart the container?

**Reply by acelaya** [reply]
https://shlink.io/documentation/install-docker-image/#use-an-external-db
